.github/workflows/smoke-checks.yml

```diff
@@ -51,8 +51,8 @@ jobs:
       run: bundle exec fastlane test device:"${{ env.IOS_SIMULATOR_DEVICE }}"
       timeout-minutes: 40
       env:
-        XCODE_VERSION: "15.2" # the most stable pair of Xcode
-        IOS_SIMULATOR_DEVICE: "iPhone 15 Pro (17.2)" # and iOS
+        XCODE_VERSION: "15.4" # the most stable pair of Xcode
```

Comment:
Temp to see if the CI passes - we might have a problem if `nonisolated(unsafe)` doesn't work on 15.2.

Reply:
we still have a problem here, need to dig into this

Reply:
I spent the whole day on this one - there's no good way. `nonisolated(unsafe)` was introduced in Swift 5.10. Tried with macros and property wrappers, but no luck. Let's discuss it if you have an idea.

Reply:
If there's no solution, we would have to drop 5.10.